### PR TITLE
azure-cli: use the same syntax than other packages

### PR DESCRIPTION
**Pull request by Et7f3**
$ nix build .#azure-cli
error: multiple hashes passed to fetchurl
(use '--show-trace' to show detailed location information)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/azure-cli/python-packages.nix [hunk 1]
```diff
@@ -539,7 +539,8 @@ let
       websocket-client = super.websocket-client.overridePythonAttrs(oldAttrs: rec {
         version = "1.3.1";
 
-        src = oldAttrs.src.override {
+        src = super.fetchPypi {
+          inherit (oldAttrs) pname;
```
Comment (Et7f3) [inline pkgs/tools/admin/azure-cli/python-packages.nix:543]: other fix is use `hash = ` instead of `sha256 = ` but I favor homogenous syntax.

**Comment by bobby285271** [comment]
@ofborg eval

**Comment by bobby285271** [comment]
I just merged #205287 since that one is opened earlier, but it should be fine to do this as well. Do you still wish to land this one?

**Comment by Et7f3** [comment]
I changed the commit to be more in sync with the title. Another advantage of this syntax is that original packages can change from sha256 to hash it will survive. So this version is more acceptable to land.

File: pkgs/tools/admin/azure-cli/python-packages.nix [hunk 2]
```diff
@@ -222,13 +222,8 @@ let
...
 
-      azure-mgmt-msi = super.azure-mgmt-msi.overridePythonAttrs (old: rec {
-        version = "6.0.1";
-        src = old.src.override {
-          inherit version;
-          sha256 = "sha256-PPkQmUoBkJ8Su7h9G2/t8dVy/PT3uCYZjlf70fnY2vU=";
-        };
-      });
+      azure-mgmt-msi = overrideAzureMgmtPackage super.azure-mgmt-msi "6.0.1" "zip"
+        "sha256-PPkQmUoBkJ8Su7h9G2/t8dVy/PT3uCYZjlf70fnY2vU=";
```
Comment (Et7f3) [inline pkgs/tools/admin/azure-cli/python-packages.nix:226]: I respect this comment https://github.com/NixOS/nixpkgs/pull/71797#discussion_r1044692955 so it should be ok

**Comment by AkechiShiro** [comment]
Hi @Et7f3 I tried again to reproduce the failure and failed to, can you close this PR unless you believe there are still fixes to insert into the latest azure-cli derivation, feel free to let me know.

**Comment by Et7f3** [comment]
Thanks for your review. Master dosesn't need this PR now.

**Pull request closed without merging**